diff --git a/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.mm b/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.mm
index 03df64d..82f8e9b 100644
--- a/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.mm
+++ b/node_modules/@react-native-community/geolocation/ios/RNCGeolocation.mm
@@ -288,7 +288,11 @@ - (void)timeout:(NSTimer *)timer
     [self enableBackgroundLocationUpdates];
 #endif
   } else if (wantsWhenInUse) {
-    [_locationManager requestWhenInUseAuthorization];
+    if(_lastUpdatedAuthorizationStatus == kCLAuthorizationStatusNotDetermined) {
+      [_locationManager requestWhenInUseAuthorization];
+    } else {
+      [self locationManagerDidChangeAuthorization:_locationManager];
+    }
   }
 }
 
