diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm
index 0d231bc..c2737d5 100644
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTPullToRefreshViewComponentView.mm
@@ -76,7 +76,9 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
     return;
   }
 
-  const auto &oldConcreteProps = static_cast<const PullToRefreshViewProps &>(*_props);
+  // Fix for RN81 regression to https://github.com/facebook/react-native/pull/49240
+  // Regression introduced by https://github.com/facebook/react-native/commit/fab7fa8
+  const auto &oldConcreteProps = static_cast<const PullToRefreshViewProps &>(*oldProps);
   const auto &newConcreteProps = static_cast<const PullToRefreshViewProps &>(*props);
 
   if (newConcreteProps.tintColor != oldConcreteProps.tintColor) {
diff --git a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
index a99f103..2893053 100644
--- a/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
+++ b/node_modules/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm
@@ -553,7 +553,6 @@ - (void)updateLayoutMetrics:(const LayoutMetrics &)layoutMetrics
   // re-applying individual sub-values which weren't changed.
   [super updateLayoutMetrics:layoutMetrics oldLayoutMetrics:_layoutMetrics];
 
-  _layoutMetrics = layoutMetrics;
   _needsInvalidateLayer = YES;
 
   _borderLayer.frame = self.layer.bounds;
@@ -575,6 +574,9 @@ - (void)updateLayoutMetrics:(const LayoutMetrics &)layoutMetrics
     auto newTransform = _props->resolveTransform(layoutMetrics);
     self.layer.transform = RCTCATransform3DFromTransformMatrix(newTransform);
   }
+
+  // Fix for https://github.com/facebook/react-native/issues/30523
+  _layoutMetrics = layoutMetrics;
 }
 
 - (BOOL)isJSResponder
